app/src/main/java/com/duckduckgo/mobile/android/duckduckgo/ui/tabswitcher/TabSwitcherPresenterImpl.java

```diff
@@ -0,0 +1,98 @@
+package com.duckduckgo.mobile.android.duckduckgo.ui.tabswitcher;
+
+import android.support.annotation.NonNull;
+
+import com.duckduckgo.mobile.android.duckduckgo.ui.browser.tab.Tab;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by fgei on 5/29/17.
+ */
+
+public class TabSwitcherPresenterImpl implements TabSwitcherPresenter {
+
+    private TabSwitcherView tabSwitcherView;
+    private List<Tab> tabs;
+    private List<Tab> tabsToDelete;
+
+    public TabSwitcherPresenterImpl() {
+        tabs = new ArrayList<>();
+        tabsToDelete = new ArrayList<>();
+    }
+
+    @Override
+    public void attach(@NonNull TabSwitcherView tabSwitcherView) {
+        this.tabSwitcherView = tabSwitcherView;
+        showTabs(tabs);
+    }
+
+    @Override
+    public void detachView() {
+        tabSwitcherView = null;
+    }
+
+    @Override
+    public void load(@NonNull List<Tab> tabs) {
+        this.tabs = tabs;
+    }
+
+    @Override
+    public void restoreState(@NonNull List<Tab> tabs, @NonNull List<Tab> tabsToRemove) {
+        this.tabsToDelete = tabsToRemove;
+        this.tabs.clear();
+        for (Tab tab : tabs) {
+            if (!tabsToDelete.contains(tab)) {
+                this.tabs.add(tab);
+            }
+        }
+    }
+
+    @Override
+    public List<Tab> saveStateTabs() {
+        return tabs;
+    }
+
+    @Override
+    public List<Tab> saveStateTabsToRemove() {
+        return tabsToDelete;
+    }
+
+    @Override
+    public void createNewTab() {
+        tabSwitcherView.resultCreateNewTab(getPositionsToDelete());
+    }
+
+    @Override
+    public void openTab(@NonNull Tab tab) {
+        tabSwitcherView.resultSelectTab(tab.index, getPositionsToDelete());
+    }
+
+    @Override
+    public void closeTab(@NonNull Tab tab) {
+        tabsToDelete.add(tab);
```

Comment:
What is your motivation for maintaining a delete list? Why not just remove it? Again a model layer would help here.